### PR TITLE
First pass for integration with current qubes-infrastructure

### PR DESCRIPTION
**Pull request by fepitre**
For now, it supports auto build for components and templates (scripts level) and qubesbuilder.TriggerRPC. Remaining RPC is done in an upcoming PR.

### CODE REVIEW

File: qubesbuilder/plugins/github/scripts/auto-build [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+
+# FIXME: update qubes-builderv2
+
+# Download component source
+./qb -c "$component" package fetch
+
+# Check if a timeout is defined
+build_timeout=$(./qb config get-var "build_timeout_$component")
+if [ -z "$build_timeout" ]; then
+    build_timeout=$(./qb config get-var build_timeout)
```
[the remark this answers is not in the document]
Comment (fepitre) [reply qubesbuilder/plugins/github/scripts/auto-build:65]: yes it's already the case on line 63.

**Comment by marmarek** [comment]
I tried (semi-successfully) setup test instance and here are issues I hit:

- [ ] missing deps in README: python3-pathspec, python3-packaging
- [ ] auto-build script prints outdated usage info (now it takes builder dir path too)
- [ ] example policy lacks `admin.vm.Kill`
- [ ] `gpg-client` option is mandatory -  can't it default to just `gpg`?
- [ ] `git -C ... tag --points-at HEAD --list v*` is called 3 times on the same(?) sources, likely once per dist (a bit painful with "qubes" executor for fetch stage too)
- [ ] remove `nosync` from mock config - fc35 vs fc32 diff too large
- [ ] "upload" isn't called automatically
- [ ] "upload" should only upload repositories that get some packages from this builder instance, otherwise it might override metadata set by another instance (for example "unstable", or "templates-...", or even individual dists - like "host" being built somewhere else than "vm");  maybe make repo list configurable in builder conf?

**Comment by fepitre** [comment]
> * "upload" should only upload repositories that get some packages from this builder instance, otherwise it might override metadata set by another instance (for example "unstable", or "templates-...", or even individual dists - like "host" being built somewhere else than "vm");  maybe make repo list configurable in builder conf?

I've improved upload by only selecting repositories that local builder is creating packages.

**Comment by fepitre** [comment]
>     * [x]  `sign`+`publish` seems to be called even if nothing was built (because no version tag) - see [Build failed: desktop-linux-manager for upload ( host:fc32 vm: bullseye fc35) marmarek/test#262](https://github.com/marmarek/test/issues/262) and [Build failed: desktop-linux-manager for upload ( host: vm: bullseye fc35) marmarek/test#263](https://github.com/marmarek/test/issues/263) (I'm not sure why those two differ? one includes `host:fc32`, and other does not)

There was an issue in `check-release-status-for-component` (missing raise).

**Pull request merged**